core_api/src/app.py

```diff
@@ -100,36 +100,40 @@ def health():
 
     return output
 
+class FileRequest(BaseModel):
+    url: AnyHttpUrl
 
 @app.post("/file", tags=["file"])
-async def create_upload_file(name: str, type: str, location: AnyHttpUrl) -> uuid.UUID:
+async def create_upload_file(file: FileRequest) -> File:
```

Comment:
Should we rename this func to `add_file` or something now that we pass by reference?

Reply:
good idea